src/components/Home/Home.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios'
+import AxiosMocker from 'axios-mock-adapter'
+
+const mockIdToken = 'anyToken'
+
+const token = {
+  idToken: mockIdToken,
+  id: 1,
+}
+
+const errorMessage = 'Error message'
+
+describe('Home', () => {
+  const mock = new AxiosMocker(axios)
+
+  afterEach(() => {
+    mock.reset()
+  })
+
+  it('should be able to return user session with correct id token', async () => {
+    const axiosGetMock = jest.spyOn(axios, 'get')
+    axiosGetMock.mockResolvedValueOnce({
+      data: { userSession: token.idToken },
+    })
+
+    const result = await axios.get('/api/sessionAuth')
+
+    expect(axiosGetMock).toHaveBeenCalledWith('/api/sessionAuth')
+    expect(result.data).toEqual({ userSession: token.idToken })
+
+    axiosGetMock.mockRestore()
+  })
+
+  it('should be able to return error during user session retrieval', async () => {
+    const axiosGetMock = jest.spyOn(axios, 'get')
+    axiosGetMock.mockRejectedValueOnce(new Error(errorMessage))
+
+    try {
```

Comment:
Pelo menos no backend, usar um try catch em testes não é uma prática recomendada, eu falaria para voce dar uma validada com a galera que manja de testes, e ver de diminuir os tamanhos de todos dos "its".


Solução que o GPT deu:
it('should be able to return error during user session retrieval', async () => {
  const axiosGetMock = jest.spyOn(axios, 'get')
  const errorMessage = 'Some error message'

  axiosGetMock.mockRejectedValueOnce(new Error(errorMessage))

  await expect(axios.get('/api/sessionAuth')).rejects.toThrow(errorMessage)
  expect(axiosGetMock).toHaveBeenCalledWith('/api/sessionAuth')

  axiosGetMock.mockRestore()
})